### PR TITLE
Create a spec for the `DAO` similar to the `Escrow`

### LINKED ISSUE #197: Create a spec for the `DAO` similar to the `Escrow`
### Motivation

The escrow has a nice [spec](https://github.com/FuelLabs/sway-applications/blob/master/escrow/SPECIFICATION.md). The DAO should have one too.

### PR DESCRIPTION
**Pull request by Braqzen**
## Type of change

- Documentation

## Changes

The following changes have been made:

- Use cases describing the functions
- Sequence diagram

## Related Issues

Closes #197

### CODE REVIEW

**Comment by Braqzen** [comment]
Leaving as open for now. `fuelup` has been updated to version 0.4 and `fmt` has either been removed, moved or something has happened to it which made CI fail when checking the formatting of other projects.

File: dao-voting/README.md [hunk 1]
```diff
@@ -15,7 +15,6 @@ More information can be found in the [specification](./SPECIFICATION.md).
 
 - The smart contract is mostly complete for the basic implementation. There are some issues that need to be worked out but a UI can be started
```
Comment (simonr0204) [inline dao-voting/README.md:16]: Do these issues have... issues? 😄

**Review by simonr0204 — APPROVED** [review]

**Comment by simonr0204** [comment]
approved pending CI passing.

File: dao-voting/README.md (hunk 1 above)
Comment (Braqzen) [reply dao-voting/README.md:16]: Yes they do. There is a bug, some design considerations and general improvements to be made in future iterations. These aren't necessarily obvious just by glancing at the current [issues](https://github.com/FuelLabs/sway-applications/issues?q=is%3Aopen+is%3Aissue+label%3A%22App%3A+DAO%22) but they do exist.

File: dao-voting/SPECIFICATION.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    - [`proposal()`](#proposal)
+    - [`governance_token_id()`](#governance_token_id)
+    - [`proposal_count()`](#proposal_count)
+- [Sequence diagram](#sequence-diagram)
+
+# Overview
+
+This document provides an overview of the application.
+
+It outline the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
```
Comment (bitzoic) [inline dao-voting/SPECIFICATION.md:24]:
```suggestion
It outlines the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
```

**Review by bitzoic — CHANGES_REQUESTED** [review]

File: dao-voting/SPECIFICATION.md (hunk 2 above)
Comment (Braqzen) [reply dao-voting/SPECIFICATION.md:24]: oh no, I think every spec might have this issue

**Review by bitzoic — APPROVED** [review]
Looks good but I would wait for @matt-user to take a look first.

**Review by matt-user — APPROVED** [review]
looks good!

**Pull request merged**